atum/src/test/scala/za/co/absa/atum/integration/SampleMeasurementsIntegrationSuite.scala

```diff
@@ -0,0 +1,131 @@
+      val outputControlMeasure: ControlMeasure = SerializationUtils.fromJson[ControlMeasure](outputData)
+
+      outputControlMeasure.metadata.additionalInfo should contain("additionalKey1", "additionalValue1")
+      outputControlMeasure.checkpoints.map(_.name) should contain theSameElementsInOrderAs Seq("Source", "Raw", "checkpoint1")
+    }
+
+    spark.disableControlMeasuresTracking()
+  }
+
+  it should "load implicit _INFO, adjust, and save implicit _INFO (stage 2)" in {
+    // Demonstrating that the default _INFO input and output will be determined automatically
+    spark.enableControlMeasuresTracking(None, None, TestBuildProperties("MySoftware", "v007")) // custom sw+version
+      .setControlMeasuresWorkflow("Job 2")
+
+    val sourceDS = spark.read
+      .parquet(s"$outputPath/stage1_job_results") // stage1_output as input for stage 2
+
+    // Some business logic of a spark job ...
+    // An example - a column rename
+    // If the renamed column is one of control measurement columns, the rename need to be registered in Control Framework
+    sourceDS.as("target")
+      .withColumnRenamed("total_response_size", "trs") // Renaming the column
+      .registerColumnRename("total_response_size", "trs") // Registering the rename, from now on the new name for the column is 'trs'
+      .filter($"trs" > 1000)
+      .setCheckpoint("checkpoint2")
+      .write.mode(SaveMode.Overwrite)
+      .parquet(s"$outputPath/stage2_job_results")
+
+    // assert the state after
+    eventually(timeout(scaled(10.seconds)), interval(scaled(500.millis))) {
+      fs.exists(new Path((s"$outputPath/stage2_job_results/_INFO"))) shouldBe true
+
+
+      val outputData = FileUtils.readFileToString(s"$outputPath/stage2_job_results/_INFO")
+      val outputControlMeasure: ControlMeasure = SerializationUtils.fromJson[ControlMeasure](outputData)
+
+      outputControlMeasure.checkpoints.map(_.name) should contain theSameElementsInOrderAs
+        Seq("Source", "Raw", "checkpoint1", "checkpoint2")
+
+      val checkpoint1 = outputControlMeasure.checkpoints.filter(_.name == "checkpoint1").head // from stage1
+      val checkpoint2 = outputControlMeasure.checkpoints.filter(_.name == "checkpoint2").head // now added
+
+      checkpoint1.software.value should startWith("atum") // default software
+      checkpoint1.version should not be empty // some default version should be present
+      checkpoint2.software.value shouldBe "MySoftware" // custom BuildProperties taking effect
+      checkpoint2.version.value shouldBe "v007"
+
+      // showing that "total_response_size" got renamed to "trs"
+      checkpoint1.controls.map(_.controlCol) should contain("total_response_size")
+      checkpoint2.controls.map(_.controlCol) should (contain("trs") and not contain ("total_response_size"))
+    }
+
+    spark.disableControlMeasuresTracking()
+  }
+}
+
+object SampleMeasurementsIntegrationSuite {
```

Comment:
I am unsure if we have an agreement on this or not, but I would rather see this at the top

Reply:
moved in https://github.com/AbsaOSS/atum/pull/106